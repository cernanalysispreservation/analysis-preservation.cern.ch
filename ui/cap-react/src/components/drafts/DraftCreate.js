import React from "react";

import Layer from "grommet/components/Layer";

import CreateForm from "./components/CreateForm";

import PropTypes from "prop-types";

class DraftCreate extends React.Component {
  render() {
    return (
      <Layer overlayClose closer flush onClose={this.props.toggle}>
        <CreateForm />
      </Layer>
    );
  }
}

DraftCreate.propTypes = {
  contentTypes: PropTypes.object,
  toggle: PropTypes.func,
  createDraft: PropTypes.func,
  formData: PropTypes.object,
  metadata: PropTypes.object
};

export default DraftCreate;
