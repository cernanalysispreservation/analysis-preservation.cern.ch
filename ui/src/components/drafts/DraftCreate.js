import React from "react";

import { connect, Provider } from "react-redux";
import store from "../../store/configureStore";

import Anchor from "grommet/components/Anchor";
import Box from "grommet/components/Box";
import Heading from "grommet/components/Heading";
import Button from "grommet/components/Button";
import Layer from "grommet/components/Layer";
import Label from "grommet/components/Label";
import Paragraph from "grommet/components/Paragraph";
import CheckmarkIcon from "grommet/components/icons/base/Checkmark";
import CloseIcon from "grommet/components/icons/base/Close";

import FormField from "grommet/components/FormField";

import { TextInput } from "grommet";
import { createDraft } from "../../actions/draftItem";

class DraftCreate extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: "",
      type: null
    };
  }

  _selectType = type => {
    this.setState({ type });
  };

  _titleChange = event => {
    let title = event.target.value;
    this.setState({ title });
  };

  _createDraft = () => {
    let { title, type } = this.state;
    let data = {
      general_title: title
    };

    this.props.createDraft(data, { name: type });
  };

  render() {
    let { history, location } = this.props;

    return (
      <Layer overlayClose closer flush onClose={this.props.toggle}>
        <Provider store={store}>
          <Box pad="large" size={{ width: "xlarge" }} wrap={false} flex>
            <Box flex={false} size="xlarge">
              <Heading tag="h4">
                Select the content type you want to create, give a title to
                distinguish from other records and start preserving!
              </Heading>
              <Box pad={{ vertical: "small" }}>
                <FormField flex={false} label="General Title">
                  <Box pad={{ horizontal: "medium" }}>
                    <TextInput onDOMChange={this._titleChange} />
                  </Box>
                </FormField>
              </Box>
              <Box pad={{ vertical: "small" }}>
                <Box>
                  <Box>
                    <Box size={{ height: { max: "medium" } }}>
                      <Box
                        direction="row"
                        wrap
                        justify="between"
                        responsive={false}
                      >
                        {this.props.contentTypes &&
                          this.props.contentTypes.map(type => (
                            <Box
                              key={type.get("deposit_group")}
                              onClick={() =>
                                this._selectType(type.get("deposit_group"))
                              }
                              basis="1/2"
                              pad="small"
                              colorIndex="light-2"
                              separator="all"
                            >
                              <Box
                                flex
                                justify="between"
                                direction="row"
                                wrap={false}
                                style={{ overflow: "visible" }}
                              >
                                <Label size="small">{type.get("name")}</Label>
                                <Box flex={false} justify="center">
                                  {type.get("deposit_group") ==
                                  this.state.type ? (
                                    <CheckmarkIcon
                                      colorIndex="ok"
                                      size="xsmall"
                                    />
                                  ) : null}
                                </Box>
                              </Box>
                            </Box>
                          ))}
                      </Box>
                    </Box>
                  </Box>
                </Box>
              </Box>
            </Box>
            <Box align="end">
              <Button
                label="Start Preserving"
                onClick={this._createDraft}
                primary={true}
              />
            </Box>
          </Box>
        </Provider>
      </Layer>
    );
  }
}

function mapStateToProps(state) {
  return {
    id: state.draftItem.get("id"),
    errors: state.draftItem.get("errors"),
    contentTypes: state.auth.getIn(["currentUser", "depositGroups"])
  };
}

function mapDispatchToProps(dispatch) {
  return {
    createDraft: (data, type) => dispatch(createDraft(data, type))
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DraftCreate);
