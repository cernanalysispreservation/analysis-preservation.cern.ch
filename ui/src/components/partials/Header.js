import React from 'react';
import PropTypes from 'prop-types';

import GrommetHeader from 'grommet/components/Header';
import Title from 'grommet/components/Title';
import Box from 'grommet/components/Box'
import Menu from 'grommet/components/Menu';
import Anchor from 'grommet/components/Anchor';
import { withRouter } from 'react-router-dom';
import {connect} from 'react-redux';
import { Link } from 'react-router-dom';

import queryString from 'query-string';
import UserIcon from 'grommet/components/icons/base/User';

import SearchBar from '../search/SearchBar';

import config from '../../config';
import { logout } from '../../actions/auth';

class Header extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <GrommetHeader fixed={false} colorIndex="neutral-1" size="small">
        <Box
          flex={true}
          pad={{horizontal: "small"}}
          justify="end"
          direction="row"
          responsive={false}>
          <Title >
            <Anchor href="#" path="/" label={config.project.name || "Project Name"} />
          </Title>
          <Box flex={true} justify="center" colorIndex="neutral-1-t">
            <SearchBar />
          </Box>
          <Menu pad={{horizontal: "small"}} direction="row" responsive={true}>
            <Anchor path="/drafts/create" label="Create" />
            <Anchor href="#" path="/search" label="Search" />
            <Menu responsive={true} icon={<UserIcon />} >
              <Anchor
                label="Logout"
                href="#"
                animateIcon={true}
                onClick={this.props.logout} />
              <Anchor
                label="Settings"
                href="#"
                animateIcon={true}
                path="/settings" />
            </Menu>
          </Menu>
        </Box>
      </GrommetHeader>
    );
  }
}

Header.propTypes = {
  logout: PropTypes.func.isRequired
};

function mapDispatchToProps(dispatch) {
  return {
    logout: () => dispatch(logout())
  };
}

export default withRouter(connect(
  ()=> ({}),
  mapDispatchToProps
)(Header));