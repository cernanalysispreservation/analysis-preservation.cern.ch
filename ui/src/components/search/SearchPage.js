import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import Box from 'grommet/components/Box';

import SearchFacets from './SearchFacets';
import SearchUtils from './SearchUtils';
import SearchResults from './SearchResults';

import {fetchSearch} from '../../actions/search';
import queryString from 'query-string';

class SearchPage extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.fetchSearch();
  }

  _changePage(page){
    let currentParams = this.getSearchParams();

    const location = {
      search: `${queryString.stringify(Object.assign(currentParams,{page: page}))}`
    };

    this.props.history.push(location);
    this.props.fetchSearch();
  }

  _changePageSize(size){
    let currentParams = this.getSearchParams();
    const location = {
      search: `${queryString.stringify(Object.assign(currentParams,{size: size}))}`
    };

    this.props.history.push(location);
    this.props.fetchSearch();
  }

  componentWillReceiveProps(nextProps) {
    if ( this.props.location.search !== nextProps.location.search ) {
      this.props.fetchSearch()
    }
  }

  getSearchParams() {
    let params = queryString.parse(this.props.location.search);
    return params;
  }

  render() {
    let utils;
    let total = null;
    let results = null;
    let aggs = null;

    let _results = {};
    let _aggs;

    if (this.props.results) {
      _results = this.props.results.toJS();
      _aggs = _results.aggregations;
    }

    if (_aggs) {
      aggs = (
        <SearchFacets
          aggs={_aggs}
          selectedAggs={this.props.selectedAggs}
          onChange={this._toggleAggs}/>
      );
    }

    if (_results && _results.hits) {
      total = _results.hits.total;
      utils = (
        <SearchUtils
          loading={this.props.loading}
          currentPage={this.props.selectedAggs.page ? parseInt(this.props.selectedAggs.page) : 1}
          size={this.props.selectedAggs.size ? parseInt(this.props.selectedAggs.size) : 10}
          total={total || 0}
          onPageChange={this._changePage}
          onPageSizeChange={this._changePageSize}
        />
      );
      results = (<SearchResults results={_results.hits.hits || []} />);
    }

    return (
      <Box  flex={true}>
        {utils}
        <Box flex={true} direction="row">
          {aggs}
          {results}
        </Box>
      </Box>
    );
  }
}

SearchPage.propTypes = {
  loading: PropTypes.bool.isRequired,
  fetchSearch: PropTypes.func,
  history: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  selectedAggs: PropTypes.array.isRequired,
  results: PropTypes.array.isRequired,
};

function mapStateToProps(state) {
  return {
    results: state.search.getIn(['results']),
    loading: state.search.getIn(['loading']),
    selectedAggs: state.search.getIn(['selectedAggs'])
  };
}

function mapDispatchToProps(dispatch) {
  return {
    fetchSearch: () => dispatch(fetchSearch()),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchPage);