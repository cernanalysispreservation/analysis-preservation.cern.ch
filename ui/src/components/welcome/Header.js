import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router";

import Anchor from "grommet/components/Anchor";
import Box from "grommet/components/Box";
import Menu from "grommet/components/Menu";
import Button from "grommet/components/Button";
import Label from "grommet/components/Label";

import Header from "grommet/components/Header";

import LoginForm from "grommet/components/LoginForm";

import { loginLocalUser } from "../../actions/auth";
import LogIcon from "grommet/components/icons/base/Login";

import CAPLogoDark from "../../img/cap-logo-dark.svg";
import CERNLogo from "./img/cern-logo.svg";

class WelcomePage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showLogin: false
    };
  }

  onFormSubmit = formData => {
    let {
      location: { state: { next: next = "/" } = {} }
    } = this.props.history;
    formData["next"] = next;

    this.props.loginLocalUser(formData);
  };

  render() {
    return (
      <Header
        pad={{ vertical: "none", horizontal: "small" }}
        size="small"
        fixed
      >
        <Box
          flex="grow"
          direction="row"
          justify="between"
          align="center"
          responsive={false}
        >
          <Box
            flex={true}
            direction="row"
            wrap={false}
            responsive={false}
            onClick={() => this.props.scrollToRef(this.props.nav.index.ref)}
          >
            <div style={{ padding: "10px 0" }}>
              <CAPLogoDark height="32px" />
            </div>
          </Box>
          <Box direction="row" align="center" responsive={false}>
            <Menu
              dropAlign={{ top: "bottom" }}
              size="small"
              direction="row"
              align="center"
              pad={{ horizontal: "medium" }}
            >
              {Object.keys(this.props.nav).map(key => (
                <Box
                  onClick={() =>
                    this.props.scrollToRef(this.props.nav[key].ref)
                  }
                >
                  <Label>{this.props.nav[key].title}</Label>
                </Box>
              ))}
            </Menu>
            {process.env.NODE_ENV === "development" ? (
              <Menu
                dropAlign={{ top: "bottom" }}
                icon={<LogIcon />}
                size="small"
                closeOnClick={false}
              >
                <LoginForm
                  usernameType="email"
                  defaultValues={{ username: "info@inveniosoftware.org" }}
                  onSubmit={this.onFormSubmit}
                />
                {this.props.authError ? (
                  <Box
                    colorIndex="critical"
                    margin={{ top: "small" }}
                    pad="small"
                  >
                    {this.props.authError}
                  </Box>
                ) : null}
              </Menu>
            ) : null}
            <Box>
              <Anchor href="/api/oauth/login/cern">
                <Box
                  flex={true}
                  separator="all"
                  direction="row"
                  wrap={false}
                  responsive={false}
                  colorIndex="neutral-1"
                  align="center"
                >
                  <Box pad={{ horizontal: "none" }} colorIndex="light-1">
                    <CERNLogo height="34px" />
                  </Box>
                  <Box pad={{ horizontal: "small" }}>
                    <Label margin="none" size="medium">
                      Login With CERN
                    </Label>
                  </Box>
                </Box>
              </Anchor>
            </Box>
          </Box>
        </Box>
      </Header>
    );
  }
}

WelcomePage.propTypes = {
  loginLocalUser: PropTypes.func.isRequired,
  authLoading: PropTypes.bool.isRequired,
  authError: PropTypes.object,
  history: PropTypes.object
};

function mapStateToProps(state) {
  return {
    authLoading: state.auth.get("loading"),
    authError: state.auth.get("error")
  };
}

function mapDispatchToProps(dispatch) {
  return {
    loginLocalUser: data => dispatch(loginLocalUser(data))
  };
}

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(WelcomePage)
);
