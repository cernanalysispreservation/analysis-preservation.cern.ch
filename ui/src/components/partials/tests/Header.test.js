import React from "react";
import { configure, shallow, render, mount } from "enzyme";
import { Map, List, fromJS } from "immutable";
import Adapter from "enzyme-adapter-react-16";
import Header from "../Header";
import Layer from "grommet/components/Layer";
import SearchPage from "../../search/SearchPage";
import DraftCreate from "../../drafts/DraftCreate";
import { MemoryRouter as Router, Switch, Route } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

configure({ adapter: new Adapter() });

let wrapper;
let store;

store = mockStore({
  auth: Map({
    isLoggedIn: true,
    currentUser: Map({
      userId: 1,
      token: 1,
      permissions: true,
      depositGroups: List()
    }),
    // token: localStorage.getItem("token"),
    error: null,
    loading: false,
    loadingInit: true,
    tokens: List(),
    integrations: Map({})
  }),
  draftItem: Map({
    errors: [],
    schemaErrors: [],
    actionsLayer: false,
    showPreviewer: false,
    filePreviewEditLayer: true,
    filePreviewEdit: {},

    workflows: fromJS([]),
    workflows_items: fromJS({}),
    bucket: Map({}),
    formData: null,
    // From backend: deposit resource
    access: null,
    can_admin: false,
    can_update: false,
    created: null,
    loading: false,
    created_by: null,
    experiment: null,
    files: Map({}),
    id: "1",
    links: null,
    metadata: {},
    revision: null,
    schema: null,
    schemas: null,
    status: null,
    type: null,
    updated: null
  })
});

describe("Main Header Component", () => {
  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Switch>
            <Header />
            <Route exact path="/search" component={SearchPage} />
          </Switch>
        </Router>
      </Provider>
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("Open How to Search page", () => {
    const shallowHeader = shallow(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );

    expect(shallowHeader).toMatchSnapshot();
  });

  it("State is updated", () => {
    expect(wrapper.state("showCreate")).not.toBeTruthy();
    wrapper.setState({ showCreate: true });
    expect(wrapper.state("showCreate")).toBeTruthy();
    expect(wrapper.state("show")).not.toBeTruthy();
    wrapper.setState({ show: true });
    expect(wrapper.state("show")).toBeTruthy();
  });

  it("Open the help layer", () => {
    // the layer in the beginning is hidden, not rendered
    expect(wrapper.find(Header).find(Layer)).toHaveLength(0);

    // const aLink = wrapper.find(".grommetux-anchor--animate-icon");
    const aLink = wrapper.find('[aria-label="circle-question"]');
    aLink.simulate("click");

    // display the layer with information regarding search
    expect(wrapper.find(Header).find(Layer)).toHaveLength(1);
  });

  it("Open the create layer", () => {
    // CreateDraft layer is hidden
    expect(wrapper.find(Header).find(DraftCreate)).toHaveLength(0);

    // create button is clicked
    const aLink = wrapper.find(".grommetux-box--clickable");
    aLink.simulate("click");

    // create draft layer is displayed
    expect(wrapper.find(Header).find(DraftCreate)).toHaveLength(1);
  });
});
