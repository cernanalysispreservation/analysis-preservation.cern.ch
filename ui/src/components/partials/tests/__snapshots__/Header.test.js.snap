// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Header Component Open How to Search page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <MemoryRouter>
        <withRouter(Connect(Header)) />
    </MemoryRouter>
</Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <MemoryRouter>
        <withRouter(Connect(Header)) />
</MemoryRouter>,
      "value": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeState": Object {
          "auth": Immutable.Map {
            isLoggedIn: true,
            currentUser: Immutable.Map {
                        userId: 1,
                        token: 1,
                        permissions: true,
                        depositGroups: Immutable.List [
                        ],
            },
            error: null,
            loading: false,
            loadingInit: true,
            tokens: Immutable.List [
            ],
            integrations: Immutable.Map {
            },
},
          "draftItem": Immutable.Map {
            can_admin: false,
            schema: null,
            access: null,
            created: null,
            bucket: Immutable.Map {
            },
            errors: Array [],
            schemaErrors: Array [],
            created_by: null,
            formData: null,
            metadata: Object {},
            schemas: null,
            status: null,
            workflows: Immutable.List [
            ],
            actionsLayer: false,
            workflows_items: Immutable.Map {
            },
            filePreviewEdit: Object {},
            files: Immutable.Map {
            },
            can_update: false,
            showPreviewer: false,
            links: null,
            type: null,
            filePreviewEditLayer: true,
            updated: null,
            id: "1",
            revision: null,
            loading: false,
            experiment: null,
},
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withRouter(Connect(Header)) />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": Object {},
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <MemoryRouter>
          <withRouter(Connect(Header)) />
</MemoryRouter>,
        "value": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "storeState": Object {
            "auth": Immutable.Map {
              isLoggedIn: true,
              currentUser: Immutable.Map {
                            userId: 1,
                            token: 1,
                            permissions: true,
                            depositGroups: Immutable.List [
                            ],
              },
              error: null,
              loading: false,
              loadingInit: true,
              tokens: Immutable.List [
              ],
              integrations: Immutable.Map {
              },
},
            "draftItem": Immutable.Map {
              can_admin: false,
              schema: null,
              access: null,
              created: null,
              bucket: Immutable.Map {
              },
              errors: Array [],
              schemaErrors: Array [],
              created_by: null,
              formData: null,
              metadata: Object {},
              schemas: null,
              status: null,
              workflows: Immutable.List [
              ],
              actionsLayer: false,
              workflows_items: Immutable.Map {
              },
              filePreviewEdit: Object {},
              files: Immutable.Map {
              },
              can_update: false,
              showPreviewer: false,
              links: null,
              type: null,
              filePreviewEditLayer: true,
              updated: null,
              id: "1",
              revision: null,
              loading: false,
              experiment: null,
},
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <withRouter(Connect(Header)) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": Object {},
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
